src/OSPSuite.InstallationValidator.Core/Domain/LogWatcher.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.IO;
+using OSPSuite.InstallationValidator.Core.Events;
+using OSPSuite.Utility;
+using OSPSuite.Utility.Events;
+
+namespace OSPSuite.InstallationValidator.Core.Domain
+{
+   public class LogWatcher : IDisposable
+   {
+      private readonly string _logFile;
+      private readonly IEventPublisher _eventPublisher;
+      private FileSystemWatcher _fileSystemWatcher;
+      private StreamReader _sr;
+
+      public LogWatcher(string logFile, IEventPublisher eventPublisher)
+      {
+         _logFile = logFile;
+         _eventPublisher = eventPublisher;
+
+         configureFileSystemWatcher(logFile);
+
+         subscribe();
+      }
+
+      private void configureFileSystemWatcher(string logFile)
+      {
+         var folderFromFileFullPath = FileHelper.FolderFromFileFullPath(logFile);
+         var fileNameEndExtension = Path.GetFileName(_logFile);
+
+         FileHelper.DeleteFile(logFile);
+         _fileSystemWatcher = new FileSystemWatcher(string.IsNullOrEmpty(folderFromFileFullPath) ? "./" : folderFromFileFullPath, fileNameEndExtension)
+         {
+            NotifyFilter = NotifyFilters.LastWrite | NotifyFilters.CreationTime | NotifyFilters.FileName | NotifyFilters.Size
+         };
+      }
+
+      private void subscribe()
+      {
+         _fileSystemWatcher.Changed += onChanged;
+         _fileSystemWatcher.Created += onCreated;
+      }
+
+      private void unsubscribe()
+      {
+         _fileSystemWatcher.Changed -= onChanged;
+         _fileSystemWatcher.Created -= onCreated;
+      }
+
+      private void onCreated(object sender, FileSystemEventArgs e)
+      {
+         var logFileStream = new FileStream(_logFile, FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
+         _sr = new StreamReader(logFileStream);
+
+         _eventPublisher.PublishEvent(new LogResetEvent(readFileText()));
+      }
+
+      private string readFileText()
+      {
+         return _sr.ReadToEnd();
+      }
+
+      private void onChanged(object sender, FileSystemEventArgs e)
+      {
+         _eventPublisher.PublishEvent(new LogAppendedEvent(readFileText()));
+      }
+
+      public void Dispose()
+      {
```

Comment:
I'd suggest to implement the dispose pattern that we are using in the Suite
you may have the template by typing `dist` followed by tab